### PR TITLE
View product details back end | James Dinh

### PR DESCRIPTION
**Pull request by jdinh8124**
Capture
Capture2
Capture3

### CODE REVIEW

File: server/index.js [hunk 1]
```diff
@@ -35,6 +35,23 @@ app.get('/api/products', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.get('/api/products/:productId', (req, res, next) => {
+  const sql = `
+  select *
+  from "products"
+  where "productId" = ${req.params.productId}
```
Comment (thebearingedge) [inline server/index.js]: This is opening you to SQL injection. Be sure to use a parameterized query whenever embedding client-supplied data in your queries.

File: server/index.js [hunk 2]
```diff
@@ -35,6 +35,23 @@ app.get('/api/products', (req, res, next) => {
...
+  `;
+  db.query(sql)
+    .then(result => {
+      if (!result.rows[0]) {
+        next(new ClientError(`cannot ${req.method} ${req.originalUrl}`, 404));
```
Comment (thebearingedge) [inline server/index.js]: You can afford to be more specific about not finding the matching product in your error message.

Comment (jdinh8124) [reply server/index.js]: Capture

File: server/index.js (hunk 1 above)
Comment (jdinh8124) [reply server/index.js]: Gotcha! 
Capture2

**Pull request merged**